public/components/trace_analytics/components/common/search_bar.tsx

```diff
@@ -46,15 +46,26 @@ export interface SearchBarProps extends FiltersProps {
 }
 
 interface SearchBarOwnProps extends SearchBarProps {
-  refresh: () => void;
+  refresh: (currService?: string, overrideQuery?: string) => Promise<void>;
   page: 'dashboard' | 'traces' | 'services' | 'app';
   datepickerOnly?: boolean;
 }
 
-export function SearchBar(props: SearchBarOwnProps) {
+export const SearchBar = forwardRef((props: SearchBarOwnProps, ref) => {
   // use another query state to avoid typing delay
   const [query, setQuery] = useState(props.query);
-  const setGlobalQuery = debounce((q) => props.setQuery(q), 50);
+  const setGlobalQuery = debounce((q) => {
+    props.setQuery(q);
+  }, 50);
+
+  // Expose a method to update the local query state
+  useImperativeHandle(ref, () => ({
+    updateQuery(newQuery: string) {
+      setQuery(newQuery);
+      setGlobalQuery(newQuery);
+      props.refresh(undefined, newQuery);
+    },
+  }));
```

Comment:
TODO for future, this can probably just be a ref from parent and attached to `EuiFieldSearch`, to simplify the logic and reduce re-renders